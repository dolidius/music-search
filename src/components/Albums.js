import DataSection from "./DataSection";
import Album from "./Album";

import { useState } from "react";
import { connect } from "react-redux";

import { getAlbumData } from "../redux/actions/album";

import AlbumDetailsModal from './AlbumDetailsModal/AlbumDetailsModal';

const Albums = ({ albums, albumDetails, getAlbumData }) => {
    const [detailsOpen, setDetailsOpen] = useState(false);
    return (
        <DataSection>

            <AlbumDetailsModal
                detailsOpen={detailsOpen}
                setDetailsOpen={setDetailsOpen}
                albumDetails={albumDetails}
            />

            {albums.results.map((album) => (
                <Album
                    key={album.collectionID}
                    album={album}
                    getAlbumData={getAlbumData}
                    setDetailsOpen={setDetailsOpen}
                />
            ))}
        </DataSection>
    );
};

const mapStateToProps = (state) => ({
    albumDetails: state.album,
});

export default connect(mapStateToProps, { getAlbumData })(Albums);
