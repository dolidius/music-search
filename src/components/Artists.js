import "../styles/main.scss";

import DataSection from './DataSection';

import Artist from './Artist';

const Artists = ({ artists }) => {
    return (
        <DataSection title="artists">
            {artists.results.map((artist) => (
                <Artist key={artist.artistId} artist={artist} />
            ))}
        </DataSection>
    );
}

export default Artists;