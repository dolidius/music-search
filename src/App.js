import { connect } from "react-redux";

import Albums from "./components/Albums";
import Search from "./components/Search";
import Artists from "./components/Artists";
import Spinner from './components/Helper/Spinner';

import { loadData } from "./redux/actions/search";

import "./styles/main.scss";

const App = ({ search, loadData }) => {
    return (
        <div className="container">
            <Search loadData={loadData} />
            {search.loading && <Spinner size={15} styles={{marginTop: '8rem'}} />}
            {!search.loading && (search.albums.resultCount > 0 || search.artists.resultCount > 0) &&(
                <div className="data-container">
                    <Albums albums={search.albums} />
                    <Artists artists={search.artists} />
                </div>
            )}
        </div>
    );
};

const mapStateToProps = (state) => ({
    search: state.searchData,
});

export default connect(mapStateToProps, { loadData })(App);
