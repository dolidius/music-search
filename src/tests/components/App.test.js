import React from "react";

import renderer from "react-test-renderer";

import App from "../../App";
import Albums from "../../components/Albums";
import Artists from "../../components/Artists";

import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore({});

describe("<App />", () => {
    const albums = [
        {
            collectionId: "1370494122",
            artworkUrl100:
                "https://is4-ssl.mzstatic.com/image/thumb/Music123/v4/01/29/19/0129190f-1a37-fb80-0518-97261962beb5/source/100x100bb.jpg",
            artistName: "frytt",
            collectionName: "Siema - Single",
        },
        {
            collectionId: "13704941232312",
            artworkUrl100:
                "https://is4-ssl.mzstatic.com/image/thumb/Music123/v4/01/29/19/0129190f-1a37-fb80-0518-97261962beb5/source/100x100bb.jpg",
            artistName: "frytt123",
            collectionName: "Siema - Single123",
        },
    ];

    const artists = [
        {
            artistName: "Guztav & Siëma",
            primaryGenreName: "Dance",
            artistId: "1",
        },
        {
            artistName: "Guztav & Siëmdasa",
            primaryGenreName: "Dance123",
            artistId: "2",
        },
    ];

    it("Should renders search bar correctly with initialState", () => {
        let store = mockStore({
            searchData: {
                albums: {},
                artists: {},
                loading: false,
            },

            album: {
                loading: false,
                albumDetails: {},
                songs: [],
                error: false,
            },
        });

        const tree = renderer
            .create(
                <Provider store={store}>
                    <App />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
    it("Should renders search bar and data with only albums", () => {
        let store = mockStore({
            searchData: {
                albums: {
                    resultCount: 2,
                    results: albums,
                },
                artists: { resultCount: 0, results: [] },
                loading: false,
            },

            album: {
                loading: false,
                albumDetails: {},
                songs: [],
                error: false,
            },
        });

        const tree = renderer
            .create(
                <Provider store={store}>
                    <App />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("Should renders search bar and data with only artists", () => {
        let store = mockStore({
            searchData: {
                albums: { resultCount: 0, results: [] },
                artists: { resultCount: 2, results: artists },
                loading: false,
            },

            album: {
                loading: false,
                albumDetails: {},
                songs: [],
                error: false,
            },
        });

        const tree = renderer
            .create(
                <Provider store={store}>
                    <App />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("Should renders search bar and data with arists and albums", () => {
        let store = mockStore({
            searchData: {
                albums: { resultCount: 2, results: artists },
                artists: { resultCount: 2, results: artists },
                loading: false,
            },

            album: {
                loading: false,
                albumDetails: {},
                songs: [],
                error: false,
            },
        });

        const tree = renderer
            .create(
                <Provider store={store}>
                    <App />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
