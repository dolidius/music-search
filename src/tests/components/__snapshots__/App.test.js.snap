// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should renders search bar and data with arists and albums 1`] = `
<div
  className="container"
>
  <form
    className="searchForm"
    onSubmit={[Function]}
  >
    <input
      className="searchForm__input"
      onChange={[Function]}
      placeholder="Search for an artist or album..."
      type="text"
      value=""
    />
    <button
      className="searchForm__submit"
      title="submitBTN"
      type="submit"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-search fa-w-16 "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </form>
  <div
    className="data-container"
  >
    <div
      className="dataSection"
      title="dataSection"
    >
      <h2
        className="dataSection__title"
      />
      <ul
        className="dataSection__list"
      >
        <li
          className="album"
          onClick={[Function]}
          title="album"
        >
          <div
            className="album__art"
          >
            <img
              alt="album artwork"
            />
          </div>
          <h3
            className="album__title"
          >
            Guztav & Siëma
             - 
          </h3>
        </li>
        <li
          className="album"
          onClick={[Function]}
          title="album"
        >
          <div
            className="album__art"
          >
            <img
              alt="album artwork"
            />
          </div>
          <h3
            className="album__title"
          >
            Guztav & Siëmdasa
             - 
          </h3>
        </li>
      </ul>
    </div>
    <div
      className="dataSection"
      title="dataSection"
    >
      <h2
        className="dataSection__title"
      >
        artists
      </h2>
      <ul
        className="dataSection__list"
      >
        <li
          className="artist"
          title="artist"
        >
          <h3
            className="artist__name"
          >
            <span
              className="artist__name--main"
            >
              Guztav & Siëma
            </span>
            <span
              className="artist__name--secondary"
            >
              Dance
            </span>
          </h3>
        </li>
        <li
          className="artist"
          title="artist"
        >
          <h3
            className="artist__name"
          >
            <span
              className="artist__name--main"
            >
              Guztav & Siëmdasa
            </span>
            <span
              className="artist__name--secondary"
            >
              Dance123
            </span>
          </h3>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<App /> Should renders search bar and data with only albums 1`] = `
<div
  className="container"
>
  <form
    className="searchForm"
    onSubmit={[Function]}
  >
    <input
      className="searchForm__input"
      onChange={[Function]}
      placeholder="Search for an artist or album..."
      type="text"
      value=""
    />
    <button
      className="searchForm__submit"
      title="submitBTN"
      type="submit"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-search fa-w-16 "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </form>
  <div
    className="data-container"
  >
    <div
      className="dataSection"
      title="dataSection"
    >
      <h2
        className="dataSection__title"
      />
      <ul
        className="dataSection__list"
      >
        <li
          className="album"
          onClick={[Function]}
          title="album"
        >
          <div
            className="album__art"
          >
            <img
              alt="album artwork"
              src="https://is4-ssl.mzstatic.com/image/thumb/Music123/v4/01/29/19/0129190f-1a37-fb80-0518-97261962beb5/source/100x100bb.jpg"
            />
          </div>
          <h3
            className="album__title"
          >
            frytt
             - 
            Siema - Single
          </h3>
        </li>
        <li
          className="album"
          onClick={[Function]}
          title="album"
        >
          <div
            className="album__art"
          >
            <img
              alt="album artwork"
              src="https://is4-ssl.mzstatic.com/image/thumb/Music123/v4/01/29/19/0129190f-1a37-fb80-0518-97261962beb5/source/100x100bb.jpg"
            />
          </div>
          <h3
            className="album__title"
          >
            frytt123
             - 
            Siema - Single123
          </h3>
        </li>
      </ul>
    </div>
    <div
      className="dataSection"
      title="dataSection"
    >
      <h2
        className="dataSection__title"
      >
        artists
      </h2>
      <ul
        className="dataSection__list"
      />
    </div>
  </div>
</div>
`;

exports[`<App /> Should renders search bar and data with only artists 1`] = `
<div
  className="container"
>
  <form
    className="searchForm"
    onSubmit={[Function]}
  >
    <input
      className="searchForm__input"
      onChange={[Function]}
      placeholder="Search for an artist or album..."
      type="text"
      value=""
    />
    <button
      className="searchForm__submit"
      title="submitBTN"
      type="submit"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-search fa-w-16 "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </form>
  <div
    className="data-container"
  >
    <div
      className="dataSection"
      title="dataSection"
    >
      <h2
        className="dataSection__title"
      />
      <ul
        className="dataSection__list"
      />
    </div>
    <div
      className="dataSection"
      title="dataSection"
    >
      <h2
        className="dataSection__title"
      >
        artists
      </h2>
      <ul
        className="dataSection__list"
      >
        <li
          className="artist"
          title="artist"
        >
          <h3
            className="artist__name"
          >
            <span
              className="artist__name--main"
            >
              Guztav & Siëma
            </span>
            <span
              className="artist__name--secondary"
            >
              Dance
            </span>
          </h3>
        </li>
        <li
          className="artist"
          title="artist"
        >
          <h3
            className="artist__name"
          >
            <span
              className="artist__name--main"
            >
              Guztav & Siëmdasa
            </span>
            <span
              className="artist__name--secondary"
            >
              Dance123
            </span>
          </h3>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<App /> Should renders search bar correctly with initialState 1`] = `
<div
  className="container"
>
  <form
    className="searchForm"
    onSubmit={[Function]}
  >
    <input
      className="searchForm__input"
      onChange={[Function]}
      placeholder="Search for an artist or album..."
      type="text"
      value=""
    />
    <button
      className="searchForm__submit"
      title="submitBTN"
      type="submit"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-search fa-w-16 "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </form>
</div>
`;
